html_to_markdown/processing.py

```diff
@@ -129,12 +130,24 @@ def _process_tag(
                 escape_misc=escape_misc,
                 escape_underscores=escape_underscores,
                 strip=strip,
+                context_before=text[-2:],
             )
 
     if tag_name and should_convert_tag:
-        return converters_map[tag_name](  # type: ignore[call-arg]
+        rendered = converters_map[tag_name](  # type: ignore[call-arg]
             tag=tag, text=text, convert_as_inline=convert_as_inline
         )
+        # For headings, ensure two newlines before if not already present
+        if is_heading:
+            prefix = ""
+            if context_before != "":
```

Comment:
```suggestion
            if context_before and not context_before.endswith("\n\n"):
```

Then reduce the redundancy below please

Reply:
Is that ok with the new version